### ISSUE TITLE
Request: external trigger recording

### ISSUE THREAD

**Issue by iceboatLen** [ccrisan/motioneyeos#842]
Hello, 
Would it be possible to add a feature where a camera will record when prompted by an external trigger on a GPIO of the Pi, such as a PIR sensor.

Thanks Ben.

**Comment by m4rky-m4rk** [comment]
I like this too. When using infra red illumination for night motion detection a big problem is the insects are attracted to IR lights causing so many false alerts outdoors IR motion detection is useless. Using PIR as a second trigger eliminates false alerts from insects... But just PIR can also cause false alerts from natural movement of leaves, wind etc. Combining both....  PIR + Motion = Recording with no false alerts :-)

**Comment by hinkma** [comment]
+1 this

**Comment by jasaw** [comment]
A quick glance at motion source code suggests that USR1 signal is supposed to trigger a movie recording, and ALRM signal supposed to record a timelapsed image. I did a quick test by send both signals to motion, but didn't seem to make any difference.
Will need time to delve into the source code to figure out why.
The function is called sig_handler, in motion.c

**Comment by ccrisan** [comment]
I remember testing that stuff with signals and it sometimes crashed and sometimes didn't work at all. I don't recall it ever working :) Anyways this was for motion 3.x. I hoped things have changed with 4.x.

**Comment by jasaw** [comment]
I hacked motion and got external trigger to work. Motion has `emulate_motion` config flag and I'm abusing it to force a recording. When USR1 signal is received, emulate_motion flag is set to 1. When USR2 signal is received, emulate_motion is set to 0. What this allows me to do is when my PIR sensor triggers, I send USR1 to motion to start recording, otherwise I send USR2 periodically to make sure emulate_motion is not left on accidentally.

Patch file attached if anyone wants to use it, just remove the txt file extension.

[motion_external_trigger_hack.patch.txt](https://github.com/ccrisan/motioneyeos/files/1043297/motion_external_trigger_hack.patch.txt)

**Comment by htrdlicn** [comment]
+1 on this request.

**Comment by jasaw** [comment]
Do all PIRs output the same signal e.g. output pulse low when there is motion? The pulse duration is not the same as the detected motion duration?
Someone who has worked with various PIR sensors please advise.
Technically, it's really simple to implement once we know the PIRs behaviour.

**Comment by htrdlicn** [comment]
They Pulse high, The two popular and inexpensive models which I have used are linked below. I currently have the SR505's wired to GPIO15 on the Pi Zero's

The behaviour is simple enough, The output pin is pulled low when there is no motion and goes high (3.3v) when motion is detected. On the SR505's this is hard wired to ~8s although there are variants that have different but fixed times. If it continues to see motion it seems to stay high, although it may be going low momentarily and then re-triggering. The SR501 has a trim pot that allows for adjustment of the pulse duration (0.3-600s). But this model is quite big.

It seems the simplest would be to start the motion event when the pin goes high and pole it once a second after that, and continue that motion event until it's low for >1s. This would accommodate either of these PIR sensors as well as dry contacts on doors quite well. 

Another use case I want to use this for is the output from a car alarm, this way installations in cars don't need to process all the motion on the street, but when the shock sensor senses something, then it can record.

HC-SR505 Mini-Body Sensor Switch 
eBay Listing 262130705424

HC-SR501 Adjust IR Pyroelectric Infrared PIR Motion
eBay Listing 252153857716

**Comment by jasaw** [comment]
I'm not familiar with motioneye code, but after a quick glance at the code, I would do something like this:

- Setup PIR handler, probably in server.py. Maybe start_motion() function...
```python
import os
import RPi.GPIO as GPIO

GPIO.setmode(GPIO.BCM)
PIR_PIN = 15
GPIO.setup(PIR_PIN, GPIO.IN)

def poll_pir():
    # to be called once per second
    if not GPIO.input(PIR_PIN):
        motion_pid = motionctl._get_pid()
        if motion_pid is not None:
            os.kill(motion_pid, signal.SIGUSR2)

def pir_motion_detected(PIR_PIN):
    motion_pid = motionctl._get_pid()
    if motion_pid is not None:
        os.kill(motion_pid, signal.SIGUSR1)

GPIO.add_event_detect(PIR_PIN, GPIO.RISING, callback=pir_motion_detected)
```

- Call poll_pir() once per second from a timer. Motioneye already uses IOLoop, maybe piggyback it on checker() function in start_motion()? I'm not familiar with IOLoop.
- Call GPIO.cleanup() in the clean up section, maybe run() function in server.py.
- Also need to install RPi.GPIO python module.
- It would be nice be able to select the GPIO number from the web front end, so some addition to main.html and main.js.

@ccrisan can definitely tell us where's an appropriate place to add PIR handler code.

**Comment by ccrisan** [comment]
This can indeed be added somewhere in `motionctl.py`. However the reason why no such functionality exists yet is that controlling motion detection by killing the daemon is just bad. We'd need to somehow use the exposed web "API" to control that. Then, exposing a command via `meyectl` would help anyone build external motion controllers by simply calling it from bash scripts or whatever.

**Comment by jasaw** [comment]
@ccrisan After having a greater understanding of motion software, I have to agree with you that it's better done via the HTTP API.

There are 2 scenarios an external trigger can be used.
1. External trigger OR image-based trigger = motion detected.
2. External trigger AND image-based trigger = motion detected.
Both scenarios can be useful as pointed out by various users here.

OR scenario can be done by hitting `http://127.0.0.1:7999/%(id)s/emulate_motion/%(enabled)s` where "enabled" can be a string "on" or "off".

AND scenario can be done by hitting `http://127.0.0.1:7999/%(id)s/detection/%(enabled)s` similar to set_motion_detection function in motionctl.py to enable/disable motion detection when an external trigger is on/off respectively. The set_motion_detection function and web UI part may have to be modified a bit.

**Comment by ccrisan** [comment]
Yeap, both scenarios are valid and useful, each for its own use case. The best way to go about it is to implement the HTTP requests in some functions in motionctl.py and expose them (1) via a meyectl command and (2) via web handlers to motionEye's UI. The former is probably the most useful for users that want to control motion themselves.

**Comment by jasaw** [comment]
@ccrisan Yes, your 2nd option would be really nice. I don't see a need for meyectl. If someone wants more complicated trigger control, they can hit motionEye's HTTP interface directly from their program.

**Comment by William1Lam** [comment]
Completely new to GIT/motioneyeos, very keen to understand the architecture and coding  of the system. However, I have not even managed to locate the Python and C source code. Where to start?

**Comment by jasaw** [comment]
@William1Lam Front-end code is [motioneye](https://github.com/ccrisan/motioneye).
Motion code [here](https://github.com/Motion-Project/motion).
Motioneyeos repo is the build system to compile all the required programs and libraries into disk images for various platforms. Motioneyeos repo does not contain the program source code.

**Comment by William1Lam** [comment]
Thank you very much for the info.

**Comment by lajo-osd** [comment]
Hi!

Any updates on this and the web handlers?

**Comment by Protarios** [comment]
On implementing such a feature, it would be perfect, if notification (mail, etc..) can be set up for each trigger type. I want to integrate motioneyeos into a doorbell, having it save a picture on motion, but only notifying if doorbell button is pushed.

**Comment by Chinchador** [comment]
I'm also looking forward to have kind of feature on MotionEye OS. 
it will be very great and for multiple uses!

**Comment by mixpc** [comment]
+1 and thank you for labeling it as 'feature request'. Look foward to seeing that web handler in motioneye

I am more inclined to the
External trigger OR image-based trigger = motion detected.
approach so that image-based trigger may be turned off in motioneye but that is a matter of personal preference as I think it would put less workload on the cpu.

**Comment by jorgekramer** [comment]
I would like to trigger a snapshot via http-request (and then download the still-image to display it in my Home-Automation-System (FHEM)). Is there a way to do this?

**Comment by ccrisan** [comment]
@jorgekramer use the Snapshot URL (which includes the required authorization signature as well).

**Comment by jorgekramer** [comment]
Tanks, it works. Is there also a way to trigger a snapshot in motioneye (e.g. if PIR in the Home-Automation-System detects motion) - and then send this image to a ftp-server?

**Comment by stefi01** [comment]
I have mine set to zero for motion sensitivaty and it detects everything but i want mine to be used at night only, the camera is raspberry pi noir v2 ir camera, also  36 x  ir 940nm leds that all turn on, gives perfect night vision in zero light for around 30feet, detects nothing at night but sees everything, the settings are all good, need a pir motion sensor to be added but how, i see lots of interest has been shown for the past year or two but no one has added this yet, why?

**Comment by jasaw** [comment]
@stefi01 If you want your PIR sensor to trigger motion detection and start recording, do this:
1. Write some python code that reads the GPIO of your PIR output.
2. On PIR trigger, run `curl "http://localhost:7999/1/config/set?emulate_motion=1"`.
3. When PIR stops triggering, run `curl "http://localhost:7999/1/config/set?emulate_motion=0"`.

**Comment by stefi01** [comment]
ok thanks for explaining, sounds easy enougth, i use my camera for night vision to see what is roaming around, i live up in mountains in california so i never know what to expect but the camera alone does not detect at all using the ir and 36 940nm leds plus another 36 on an add on board so hoping the pir works better, if i get stuck i will ask again and show what i have done but if it works then i could post those results too as it may help others :)

**Comment by schnabelnator** [comment]
@jasaw could you point me into a direction for the python code? Im running a raspberry pi 3b+ using the motion distro, do i have to use another distro with motion on top of it to be able to use the gpio pins?

Thank you!

**Comment by jasaw** [comment]
Something like this that polls the GPIO pin (assuming active low PIR signal). You run this as a standalone program. If you are running motionEyeOS, you can run this program from `/data/etc/userinit.sh`.
Change the `PIR_GPIO_PIN` to whichever pin you have connected to your PIR.
Change the background flag to `True` if you want it to daemonize (needed if running from `userinit.sh`).

```python
#!/usr/bin/python
""" Watches GPIO pin for external motion trigger """

import sys
import os
import signal
import time
import cStringIO
import pycurl
import RPi.GPIO as GPIO

terminate = False
PIR_GPIO_PIN = 17

def handle_signals(signum, stack):
    global terminate
    if signum == signal.SIGTERM or signum == signal.SIGINT:
        terminate = True

def createDaemon():
    UMASK = 0
    WORKDIR = "/"
    MAXFD = 1024
    REDIRECT_TO = "/dev/null"
    if (hasattr(os, "devnull")):
        REDIRECT_TO = os.devnull

    try:
        pid = os.fork()
    except OSError, e:
        raise Exception, "%s [%d]" % (e.strerror, e.errno)
    if (pid == 0):
        os.setsid()
        try:
            pid = os.fork()
        except OSError, e:
            raise Exception, "%s [%d]" % (e.strerror, e.errno)

        if (pid == 0):
            os.chdir(WORKDIR)
            os.umask(UMASK)
        else:
            os._exit(0)
    else:
        os._exit(0)

    import resource
    maxfd = resource.getrlimit(resource.RLIMIT_NOFILE)[1]
    if (maxfd == resource.RLIM_INFINITY):
        maxfd = MAXFD
    for fd in range(0, maxfd):
        try:
            os.close(fd)
        except OSError:
            pass
    os.open(REDIRECT_TO, os.O_RDWR)
    os.dup2(0, 1)
    os.dup2(0, 2)
    return(0)


def http_req(url):
    curl = pycurl.Curl()
    curl.setopt(pycurl.URL, url)
    response = cStringIO.StringIO()
    curl.setopt(pycurl.WRITEFUNCTION, response.write)
    response_headers = []
    curl.setopt(pycurl.HEADERFUNCTION, response_headers.append)
    try:
        curl.perform()
    except pycurl.error, e:
        print str(e)
    response_code = curl.getinfo(pycurl.RESPONSE_CODE)
    response_type = curl.getinfo(pycurl.CONTENT_TYPE)
    curl.close()
    response_data = response.getvalue()
    response.close()
    return response_code


def run():
    global terminate
    background = False

    # TODO: look for 'daemonize' and 'pir gpio pin' options from command line arguments

    if background:
        createDaemon()

    signal.signal(signal.SIGTERM, handle_signals)
    signal.signal(signal.SIGINT, handle_signals)

    GPIO.setmode(GPIO.BCM)
    GPIO.setwarnings(False)
    GPIO.setup(PIR_GPIO_PIN, GPIO.IN,pull_up_down = GPIO.PUD_UP)

    GPIO.input(PIR_GPIO_PIN)

    prev_state = None
    while not terminate:
        time.sleep(0.5)
        cur_state = GPIO.input(PIR_GPIO_PIN)
        if cur_state == False:
            url = 'http://localhost:7999/1/config/set?emulate_motion=1'
        elif cur_state == True:
            url = 'http://localhost:7999/1/config/set?emulate_motion=0'
        if prev_state != cur_state:
            rc = http_req(url)
            print rc
            prev_state = cur_state


if __name__ == "__main__":
    run()
```

**DISCLAIMER:** I haven't tested this code, use at your own risk.

**Comment by stefi01** [comment]
@jasaw  where would the code go if it is MotionEye installed along side Raspbian and I guess we would need the above code and some  other code that reads the GPIO pin

**Comment by stefi01** [comment]
@jasaw i forgot to ask, what do i name the file, i hate to ask but still fairly new to raspberry pi

**Comment by stefi01** [comment]
@jasaw  i think im nearly there, got the file set up, got the curl start n stop lines in morione eye run comand setion. im not where to execute file permission, like where do i do that and with what code and also i checked in rc.local but everything there is commented out so not sure what to put in that, apart from that i think i am all setup

**Comment by stefi01** [comment]
might have worked out the rc.local part, here is my my file after i added the line, the rest was already there , i think it is right, just not sure now for the where to add the permission part

#
# By default this script does nothing.

# Print the IP address
_IP=$(hostname -I) || true
if [ "$_IP" ]; then
  printf "My IP address is %s\n" "$_IP"
fi

sudo python /home/pi/pir_reader &
exit 0

**Comment by jasaw** [comment]
@stefi01 
To enable execute permission, run `chmod +x pir_reader`.
Did you change the background flag to True in the code?
You should be able just to this:

```
#
# By default this script does nothing.

# Print the IP address
_IP=$(hostname -I) || true
if [ "$_IP" ]; then
  printf "My IP address is %s\n" "$_IP"
fi

/home/pi/pir_reader
exit 0
```

**Comment by stefi01** [comment]
ok gotcha, gona run the camera tonight outside as trying to find out what is trying to eat  live wires and that, have it working in night mode with a raspberry pi noir and 2x 32 ir led ring's, detects pretty well up to about 12 feet but can see with out light for around 50feet, not bad but the pir should add an extra 8 to 9 feet, tommorow i will play around with the  pir sensor, i redesigned someone elses camera file on thingiverse and printed a nice camera just for MotionEye :)

**Comment by schnabelnator** [comment]
Thank you very much for your help @jasaw!
I have saved your code above under /data/etc/motionDetection.py and my /data/etc/userinit.sh only has those two lines:

```
/home/pi/motionDetection.py
exit 0
```
Is this correct?
I used chmod to make the motionDetection.py executable and changed the backgroundflag to true. However the script does not seem to work!
Is there any way to monitor it's output or see if it is running under MotionOS?

Thank you very much!

**Comment by jasaw** [comment]
@schnabelnator Use upper case for background flag `True`. 
To debug, change the background flag back to `False` (note the upper case), then run the script `/home/pi/motionDetection.py` from command line and it will run in the foreground. It should print "200" when there's a change in the PIR GPIO pin.

**Comment by schnabelnator** [comment]
@jasaw Thank you for your help!
I have changed your suggested code to work with 3 diffrent input GPIOs, if one is high then all 3 cameras should start recording. A text ("motionDetected") is supposed to be displayed on the camera that corresponds to the alarm. My Code does work but only once, the while loop is immediately stopped and I can't figure out why. If I use your code it works but with my changes it does not. Could you take a glance at my code? This is my first python script...
Thank you!

```
#!/usr/bin/python
""" Watches GPIO pin for external motion trigger """

import sys
import os
import signal
import time
import cStringIO
import pycurl
import RPi.GPIO as GPIO

terminate = False
ALARM_GPIO_PIN_1 = 7
ALARM_GPIO_PIN_2 = 11
ALARM_GPIO_PIN_3 = 13

def handle_signals(signum, stack):
    global terminate
    if signum == signal.SIGTERM or signum == signal.SIGINT:
        terminate = True

def createDaemon():
    UMASK = 0
    WORKDIR = "/"
    MAXFD = 1024
    REDIRECT_TO = "/dev/null"
    if (hasattr(os, "devnull")):
        REDIRECT_TO = os.devnull

    try:
        pid = os.fork()
    except OSError, e:
        raise Exception, "%s [%d]" % (e.strerror, e.errno)
    if (pid == 0):
        os.setsid()
        try:
            pid = os.fork()
        except OSError, e:
            raise Exception, "%s [%d]" % (e.strerror, e.errno)

        if (pid == 0):
            os.chdir(WORKDIR)
            os.umask(UMASK)
        else:
            os._exit(0)
    else:
        os._exit(0)

    import resource
    maxfd = resource.getrlimit(resource.RLIMIT_NOFILE)[1]
    if (maxfd == resource.RLIM_INFINITY):
        maxfd = MAXFD
    for fd in range(0, maxfd):
        try:
            os.close(fd)
        except OSError:
            pass
    os.open(REDIRECT_TO, os.O_RDWR)
    os.dup2(0, 1)
    os.dup2(0, 2)
    return(0)


def http_req(url):
    curl = pycurl.Curl()
    curl.setopt(pycurl.URL, url)
    response = cStringIO.StringIO()
    curl.setopt(pycurl.WRITEFUNCTION, response.write)
    response_headers = []
    curl.setopt(pycurl.HEADERFUNCTION, response_headers.append)
    try:
        curl.perform()
    except pycurl.error, e:
        print str(e)
    response_code = curl.getinfo(pycurl.RESPONSE_CODE)
    response_type = curl.getinfo(pycurl.CONTENT_TYPE)
    curl.close()
    response_data = response.getvalue()
    response.close()
    return response_code


def run():
    global terminate
    background = False

    # TODO: look for 'daemonize' and 'pir gpio pin' options from command line arguments

    if background:
        createDaemon()

    signal.signal(signal.SIGTERM, handle_signals)
    signal.signal(signal.SIGINT, handle_signals)

    GPIO.setmode(GPIO.BOARD)
    GPIO.setwarnings(False)
    GPIO.setup(ALARM_GPIO_PIN_1, GPIO.IN,pull_up_down = GPIO.PUD_DOWN)
    GPIO.setup(ALARM_GPIO_PIN_2, GPIO.IN,pull_up_down = GPIO.PUD_DOWN)
    GPIO.setup(ALARM_GPIO_PIN_3, GPIO.IN,pull_up_down = GPIO.PUD_DOWN)

    alarm_1 = GPIO.input(ALARM_GPIO_PIN_1)
    alarm_2 = GPIO.input(ALARM_GPIO_PIN_2)
    alarm_3 = GPIO.input(ALARM_GPIO_PIN_3)
    print alarm_1
    print alarm_2
    print alarm_3

    prev_state = None
    while not terminate:
        time.sleep(0.5)
        cur_state = alarm_1 or alarm_2 or alarm_3
        if cur_state == False:
            url_1 = 'http://localhost:7999/1/config/set?emulate_motion=0'
            url_2 = 'http://localhost:7999/2/config/set?emulate_motion=0'
            url_3 = 'http://localhost:7999/3/config/set?emulate_motion=0'
            url_4 = 'http://localhost:7999/1/config/set?text_left="%20"'
            url_5 = 'http://localhost:7999/2/config/set?text_left="%20"'
            url_6 = 'http://localhost:7999/3/config/set?text_left="%20"'
        elif cur_state == True:
            url_1 = 'http://localhost:7999/1/config/set?emulate_motion=1'
            url_2 = 'http://localhost:7999/2/config/set?emulate_motion=1'
            url_3 = 'http://localhost:7999/3/config/set?emulate_motion=1'
            if alarm_1 == True:
		print 'alarm1'
                url_4 = 'http://localhost:7999/1/config/set?text_left="motionDetected"'
                url_5 = 'http://localhost:7999/2/config/set?text_left="%20"'
                url_6 = 'http://localhost:7999/3/config/set?text_left="%20"'
            if alarm_2 == True:
		print 'alarm2'
                url_4 = 'http://localhost:7999/1/config/set?text_left="%20"'
                url_5 = 'http://localhost:7999/2/config/set?text_left="motionDetected"'
                url_6 = 'http://localhost:7999/3/config/set?text_left="%20"'
            if alarm_3 == True:
		print 'alarm3'
                url_4 = 'http://localhost:7999/1/config/set?text_left="%20"'
                url_5 = 'http://localhost:7999/2/config/set?text_left="%20"'
                url_6 = 'http://localhost:7999/3/config/set?text_left="motionDetected"'
            
        if prev_state != cur_state:
            rc_1 = http_req(url_1)
            rc_2 = http_req(url_2)
            rc_3 = http_req(url_3)
            rc_4 = http_req(url_4)
            rc_5 = http_req(url_5)
            rc_6 = http_req(url_6)
            print rc_1
            print rc_2
            print rc_3
            prev_state = cur_state

if __name__ == "__main__":
    run()

```

**Comment by mdeviprasad** [comment]
Is it possible to add a LCD so that it can show time .... The idea is to make it look like a watch, leave it aside in a room  ... newbie hear .. have no idea of programming ...

**Comment by jasaw** [comment]
@mdeviprasad Sure, but most likely you'll have to write some code to write to the I2C bus (`/dev/i2c-1`). What to write depends on your LCD. Read your LCD datasheet.

**Comment by joland** [comment]
I am also looking for a simple GPIO doorbell on/off button hookup. IFTTT offers a straight forward webhook option. All i need is motion eye os to accept external GPIO connections and 1 line of code to connect to IFTTT with curl...

**Comment by judalvarezca** [comment]
Hello! Thanks for the great work.
A quick question: How should I build the URL for the HTTP request? I have tried with `url = 'http://localhost:7999/1/config/set?emulate_motion=0'` as you did in your example but it has no success. The port is the motioneye port? I keep receiving `{"error": "not found"}` as response.
Have a great day.

**Comment by ccrisan** [comment]
Port 7999 is, by default, the Motion port. 80 or 8765 is used by motionEye. The funny thing is that you got an error from motionEye, accessing the Motion port. I can see the following possibilities:

* you configured motionEye to run on port 7999
* you mistyped the command and you're actually somehow requesting on port 80

**Comment by judalvarezca** [comment]
Thank you for the quick response. 
Digging into my configuration files, I found that my motion config is: 

`# @enabled on`
`# @show_advanced on`
`# @normal_password `
`# @admin_username admin`
`# @admin_password `
`# @normal_username user`


`webcontrol_html_output on`
`webcontrol_port 8999`
`setup_mode off`
`webcontrol_parms 2`
`webcontrol_localhost off`
`armstate false`
`lightstate false`

I setted the `webcontrol_localhost off` to access from any machine to the motion port. However, when I do the request I receive "`curl: (7) Failed to connect to localhost port 8999: Connection Refused`". What am I missing here?

**Comment by judalvarezca** [comment]
Nevermind, I was trying to set the motion of a remote motionEye camera into my local server. Silly me. When I used the remote IP everything works fine. Thanks for all the support, always appreciated.

**Comment by peladam** [comment]
> 
> 
> @stefi01 If you want your PIR sensor to trigger motion detection and start recording, do this:
>     1. Write some python code that reads the GPIO of your PIR output.
>     2. On PIR trigger, run `curl "http://localhost:7999/1/config/set?emulate_motion=1"`.
>     3. When PIR stops triggering, run `curl "http://localhost:7999/1/config/set?emulate_motion=0"`.

Hello there,
I am getting a slight problem while using those urls to trigger recording. I am getting a corrupted videos which are about 330 kB. "Normal" recording (while camera detects motion) works perfect ^_^. 

Here are mine logs:
motion.log
https://pastebin.com/NMjWpdrX
motioneye.log
https://pastebin.com/JuDCtjMD
messages.log
https://pastebin.com/RF4k48xm
boot.log
https://pastebin.com/qumW9vvz
dmesg.log
https://pastebin.com/KT8nrmkH

**Comment by vinceemail** [comment]
The script above is great! I'm using a PIR to trigger recording, and stop recording when the motion stops. This works well, BUT ONLY when motion detection is turned on. 

This means that the Pi is actively processing the video stream, and detecting motion in the image.

I believe this is using up unnecessary processor bandwidth, in my case, because I would like to rely solely on the PIR sensor. 

Is there a way to **disable** camera based detection, but still allow "emulate_motion"?

Many thanks :)

**Comment by jasaw** [comment]
@vinceemail That requires changes in the motion software. I agree that it would be nice to have an independent "forced recording" mode, rather than depend on "emulate_motion". Can you please create a new feature request in [motion](https://github.com/Motion-Project/motion) project?

**Comment by joseluissegura** [comment]
have you @vinceemail @jasaw tried?
using: 
 http://192.168.1.14:7999/1/detection/start
 http://192.168.1.14:7999/1/detection/pause

where 192.168.1.14:7999 is the ip:port of your rpi 

refer to issue #502

**Comment by joseluissegura** [comment]
another possibility could be that the PIR set the camera to active processing the video stream
and then back to "pause" 

reducing the false positives and the heating of the PI due to the active processing of videos all the time

**Comment by vpmotion** [comment]
@jasaw @ccrisan 
Hi
I wonder if you could possibly help me. I currently have 3 raspberry pi zero with the motioneyeos installed and all 3 configured as fast network camera, I then have a raspberry pi 3b+ with motioneye installed alongside the raspberian buster. I have put the above code for external trigger on the pi 3b+ and works with no issues. What i would like to know is there a way that i could get the external trigger and the working schedule to work together. 

The problem i am facing is that when i set the schedule to not trigger at certain times and i send a external trigger the motioneye still records weather or not i have set the working schedule.

any and all help would be much appreciated.

**Comment by vpmotion** [comment]
Can someone point me to where the schedule part of the code is. I think I might need to somehow add a bit of extra code there to pause or stop any readings coming in from the GPIO pins?

**Comment by vpmotion** [comment]
Hi

Can anyone please help with the above I would really really appreciate all the help or direction I can get please.

**Comment by jasaw** [comment]
@vpmotion Look here for working schedule function: https://github.com/ccrisan/motioneye/blob/78a89fbc2e2d2c284408f8eb1037c8598b4e715a/motioneye/wsswitch.py#L34

**Comment by vpmotion** [comment]
@Jasaw

Thanks for that. Would there be a way to get the wsswitch to run a script instead of triggering the motion detection on and off?

**Comment by vinceemail** [comment]
> have you @vinceemail @jasaw tried?
> using:
> http://192.168.1.14:7999/1/detection/start
> http://192.168.1.14:7999/1/detection/pause
> 
> where 192.168.1.14:7999 is the ip:port of your rpi
> 
> I saw that while http://192.168.1.14:7999/1/detection/status is "pause" the
> 
> http://192.168.1.14:7999/1/config/set?emulate_motion=1
> http://192.168.1.14:7999/1/config/set?emulate_motion=0
> 
> where able to execute correctly saving pictures, movies, sending emails etc
> 
> saving pictures only when emulate_motion is set to 1
> and not when the the Pi is actively processing the video stream, and detecting motion in the image

What a great idea!! I'll give it a shot! Thank you! :) 

Right now, I am actually using a really simple hand written script, which just records and uploads a video to google drive while PIR motion is detected. It's working really well, with flawless framerates, even on a PiZero (that always seemed to be an issue for me on MotionEyeOS). 

But, I'll give this a shot! 

Thanks again!

**Comment by jasaw** [comment]
@vinceemail Thank you for sharing. That is an easier way of doing it.

**Comment by smcgann99** [comment]
@vinceemail
I have tried the using the above and seem to get correct responses eg emulate_motion = 1 Done 
but i do not see any files created, any ideas please?

**Comment by jedtrently** [comment]
> 
> 
> have you @vinceemail @jasaw tried?
> using:
> http://192.168.1.14:7999/1/detection/start
> http://192.168.1.14:7999/1/detection/pause
> 
> where 192.168.1.14:7999 is the ip:port of your rpi
> 
> I saw that while http://192.168.1.14:7999/1/detection/status is "pause" the
> 
> http://192.168.1.14:7999/1/config/set?emulate_motion=1
> http://192.168.1.14:7999/1/config/set?emulate_motion=0
> 
> where able to execute correctly saving pictures, movies, sending emails etc
> 
> saving pictures only when emulate_motion is set to 1
> and not when the the Pi is actively processing the video stream, and detecting motion in the image

What I would like to do is constantly stream using MotionEye whilst using a PIR sensor connected to the GPIO to trigger video recording for a certain timeframe AFTER motion is detected, like how Motion does with the post_capture and post_gap settings.

Is this possible by using the above method? Or another suggested by @jasaw ?

Thanks! I'd really appreciate the help, I'm just learning RPi and just have the last piece of the puzzle to make my Bird box stream work!! :)

**Comment by avanc** [comment]
I added @jasaw 's script with some modifications to a git repository to have all the information spread across issues (https://github.com/ccrisan/motioneye/issues/31) at a central place:
https://github.com/avanc/motioneye-pir

Feedback is always welcome!

Interestingly, it also works if motion detection is deactivated within motionEye. This is in contrast to the observations at https://github.com/ccrisan/motioneyeos/issues/842#issuecomment-513788717

**Comment by jedtrently** [comment]
> 
> 
> I added @jasaw 's script with some modifications to a git repository to have all the information spread across issues ([ccrisan/motioneye#31](https://github.com/ccrisan/motioneye/issues/31)) at a central place:
> https://github.com/avanc/motioneye-pir
> 
> Feedback is always welcome!
> 
> Interestingly, it also works if motion detection is deactivated within motionEye. This is in contrast to the observations at [#842 (comment)](https://github.com/ccrisan/motioneyeos/issues/842#issuecomment-513788717)

Thank you very much! I will definitely implement this when the current system is renewed!!

**Comment by enry86cami** [comment]
> @stefi01 If you want your PIR sensor to trigger motion detection and start recording, do this:
> 
> 1. Write some python code that reads the GPIO of your PIR output.
> 2. On PIR trigger, run `curl "http://localhost:7999/1/config/set?emulate_motion=1"`.
> 3. When PIR stops triggering, run `curl "http://localhost:7999/1/config/set?emulate_motion=0"`.

Hello, I'm trying to start a recording with: curl "http://192.168.0.10:8085/1/config/set?emulate_motion=1" 
where 192.168.0.10 is the ip of my raspberry running motioneye and 8085 is the port of the camera.
In motioneye I configured the file storage root directory: /home/pi/mnt/video_nas/Sala and the motioneye created correctly the folder. 
After that I "switched on" the "movies" ; under "record mode" I selected: motion triggered

By cmd on my pc I type: curl "http://192.168.0.10:8085/1/config/set?emulate_motion=1" and I receive:
"Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file."

And nothing happens in the folder....

**Comment by starbasessd** [comment]
The control port is 7999. The stream port is 8085. To turn on or off functions you need to use port 7999.

**Comment by enry86cami** [comment]
> The control port is 7999. The stream port is 8085. To turn on or off functions you need to use port 7999.

thanks for reply, ok but how can I choose which camera start recording?
 if I try: curl "http://192.168.0.10:7999/1/config/set?emulate_motion=1" i receive:
curl: (7) Failed to connect to 192.168.0.10 port 7999: Connection refused

**Comment by starbasessd** [comment]
https://motion-project.github.io/motion_config.html#OptDetail_Webcontrol
curl http://IP:7999/camera#/command
for your case (assuming camera 5):
{IP}:{port}/{camid}/action/eventstart          #Trigger a new event.
curl "http://192.168.0.10:7999/5/eventstart"
to end:
curl "http://192.168.0.10:7999/5/eventend"

**Comment by enry86cami** [comment]
> https://motion-project.github.io/motion_config.html#OptDetail_Webcontrol
> curl http://IP:7999/camera#/command
> for your case (assuming camera 5):
> {IP}:{port}/{camid}/action/eventstart #Trigger a new event.
> curl "http://192.168.0.10:7999/5/eventstart"
> to end:
> curl "http://192.168.0.10:7999/5/eventend"

I checked and the control port is:
```
# the TCP port that motion HTTP control interface listens on
motion_control_port 7999
```

but for example if I type:
curl "http://192.168.0.10:7999/2/action/eventstart"

the reply is:
curl: (7) Failed to connect to 192.168.0.10 port 7999: Connection refused

so I suppose that the command didn't been receive

**Comment by starbasessd** [comment]
Are you running motionEye or motionEyeOS?

**Comment by enry86cami** [comment]
> Are you running motionEye or motionEyeOS?

motioneye

**Comment by starbasessd** [comment]
Edit /etc/motioneye/motion.conf:
set 
webcontrol_port 7999
setup_mode off
webcontrol_parms 2
webcontrol_localhost off
webcontrol_interface 1
and save
restart motioneye or reboot.
Web interface for motion (port 7999) should now be active for computers on local network.

**Comment by enry86cami** [comment]
> Edit /etc/motioneye/motion.conf:
> set
> webcontrol_port 7999
> setup_mode off
> webcontrol_parms 2
> webcontrol_localhost off
> webcontrol_interface 1
> and save
> restart motioneye or reboot.
> Web interface for motion (port 7999) should now be active for computers on local network.

is the same... this is the file:
`

**Comment by starbasessd** [comment]
With it set correctly if I open a browser and point to:
http://ip_address:7999
I get:
Motion 4.2.2 Running [6] Cameras 
1 
2 
3 
4 
5 
6

**Comment by enry86cami** [comment]
> http://ip_address:7999
I also rebooted raspberry
![image](https://user-images.githubusercontent.com/54316185/133629884-4bac4652-e3d2-4fc3-9f07-183df4687eb8.png)

**Comment by starbasessd** [comment]
If you didn't follow the install instructions from here:
https://github.com/ccrisan/motioneye/wiki/Installation
You may also need to edit /etc/motion/motion.conf

**Comment by starbasessd** [comment]
Are you running a firewall on the 192.168.0.10?
What OS are you running? Maybe Arch, or Gentoo, or something really locked down?

**Comment by enry86cami** [comment]
> Are you running a firewall on the 192.168.0.10?
> What OS are you running? Maybe Arch, or Gentoo, or something really locked down?

It is instelled on a raspberry 3b with raspbian, no firewall...

**Comment by starbasessd** [comment]
Desktop or Lite?

**Comment by enry86cami** [comment]
> If you didn't follow the install instructions from here:
> https://github.com/ccrisan/motioneye/wiki/Installation
> You may also need to edit /etc/motion/motion.c

> Desktop or Lite?

desktop

**Comment by starbasessd** [comment]
Open the local browser on the Pi, and check http://localhost:7999 and
in terminal try running the curl commands above...

**Comment by starbasessd** [comment]
I am building a Pi3B+ right now, latest Raspbian and updates and instructions from Wiki, with the changes I listed above.

**Comment by enry86cami** [comment]
> Open the local browser on the Pi, and check http://localhost:7999 and
> in terminal try running the curl commands above...

On the raspberry localhost:7999 works!
 but not 192.168.0.10:7999.....

**Comment by starbasessd** [comment]
What does command 
ip a
show in terminal on the Pi?

**Comment by enry86cami** [comment]
pi@raspberrypi:~ $ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether b8:27:eb:ff:88:8b brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.10/24 brd 192.168.0.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::63aa:eb2e:6730:c243/64 scope link 
       valid_lft forever preferred_lft forever
3: wlan0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast state DOWN group default qlen 1000
    link/ether b8:27:eb:aa:dd:de brd ff:ff:ff:ff:ff:ff

**Comment by starbasessd** [comment]
OK, interesting it can't find itself via IP.
Can you ping 192.168.0.10?

**Comment by enry86cami** [comment]
> OK, interesting it can't find itself via IP.
> Can you ping 192.168.0.10?

yes I can ping from raspberry. I can ping from pc pc also...

in the same raspberry runs a local BLYNK server and I didn't have similar problems

**Comment by starbasessd** [comment]
what port(s) does BLYNK use?

**Comment by enry86cami** [comment]
honestly I don't know... If I change the webcontrol_port from 7999 to another? for example 7888?

**Comment by starbasessd** [comment]
I missed a setting:
edit /etc/motioneye/motioneye.conf
change 
\# whether motion HTTP control interface listens on
\# localhost or on all interfaces
motion_control_localhost false

\# the TCP port that motion HTTP control interface listens on
motion_control_port 7999

Sorry, lack of caffeine.

**Comment by starbasessd** [comment]
restart motioneye or reboot...

**Comment by starbasessd** [comment]
Built a clean new Desktop image of RaspberryPiOS (2021/05/07 start) with all patches and updates.
Installed motionEye from instructions at: 
https://github.com/ccrisan/motioneye/wiki/Install-On-Raspbian
Edited /etc/motioneye/motioneye.conf
changed: motion_control_localhost true   to  motion_control_localhost false and restarted motioneye.
Can now send curl commands from both local network PCs and desktop.
If I connect from local network PCs or Pi desktop browser I get:
Motion 4.3.2 Running [2] Cameras 
1 
2

**Comment by enry86cami** [comment]
> I missed a setting:
> edit /etc/motioneye/motioneye.conf
> change
> # whether motion HTTP control interface listens on
> # localhost or on all interfaces
> motion_control_localhost false
> 
> # the TCP port that motion HTTP control interface listens on
> motion_control_port 7999
> 
> Sorry, lack of caffeine.


I tried to change the port to 7888 also but nothing:
`
# path to the configuration directory (must be writable by motionEye)
conf_path /etc/motioneye

# path to the directory where pid files go (must be writable by motionEye)
run_path /var/run

# path to the directory where log files go (must be writable by motionEye)
log_path /var/log

# default output path for media files (must be writable by motionEye)
media_path /var/lib/motioneye

# the log level (use quiet, error, warning, info or debug)
log_level info

# the IP address to listen on
# (0.0.0.0 for all interfaces, 127.0.0.1 for localhost)
listen 0.0.0.0

# the TCP port to listen on
port 8765

# path to the motion binary to use (automatically detected if commented)
#motion_binary /usr/bin/motion

# whether motion HTTP control interface listens on
# localhost or on all interfaces
motion_control_localhost false

# the TCP port that motion HTTP control interface listens on
motion_control_port 7888

# interval in seconds at which motionEye checks if motion is running
motion_check_interval 10

# whether to restart the motion daemon when an error occurs while communicating with it
motion_restart_on_errors false

# interval in seconds at which motionEye checks the SMB mounts
mount_check_interval 300

# interval in seconds at which the janitor is called
# to remove old pictures and movies
cleanup_interval 43200

# timeout in seconds to wait for response from a remote motionEye server
remote_request_timeout 10

# timeout in seconds to wait for mjpg data from the motion daemon
mjpg_client_timeout 10

# timeout in seconds after which an idle mjpg client is removed
# (set to 0 to disable)
mjpg_client_idle_timeout 10

# enable SMB shares (requires motionEye to run as root) 
smb_shares false

# the directory where the SMB mount points will be created
smb_mount_root /media

# path to the wpa_supplicant.conf file
# (enable this to configure wifi settings from the UI)
#wpa_supplicant_conf /etc/wpa_supplicant.conf

# path to the localtime file
# (enable this to configure the system time zone from the UI)
#local_time_file /etc/localtime

# enables shutdown and rebooting after changing system settings
# (such as wifi settings or time zone)
enable_reboot false

# timeout in seconds to use when talking to the SMTP server
smtp_timeout 60

# timeout in seconds to wait for media files list
list_media_timeout 120

# timeout in seconds to wait for media files list, when sending emails
list_media_timeout_email 10

# timeout in seconds to wait for zip file creation
zip_timeout 500

# timeout in seconds to wait for timelapse creation
timelapse_timeout 500

# enable adding and removing cameras from UI
add_remove_cameras true

# enables HTTP basic authentication scheme (in addition to, not instead of the signature mechanism)
http_basic_auth false

# overrides the hostname (useful if motionEye runs behind a reverse proxy)
# server_name motionEye


webcontrol_port 7888
setup_mode off
webcontrol_parms 2
webcontrol_localhost off
webcontrol_interface 1
`

I don't know why there is this formatting....

**Comment by starbasessd** [comment]
I don't know. Works on my Production motionEye hub, and several test machines ( motionEye on other OSs, and RaspberryPiOS on various hardware. 
Can you run command:
netstat -apt
and report the result?

If not, my suggestion would be to start clean on a different USB or SDCard (save this setup)
install RPiOS, update everything, and then just install motionEye ONLY.
This would eliminate the BLYNK as being interfering, then re-add BLYNK after confirming motionEye is working.

As to the freaky text formatting, MarkUp (here in Github) thinks '\#' as the first character on a line designates 
# BOLD

**Comment by enry86cami** [comment]
> I don't know. Works on my Production motionEye hub, and several test machines ( motionEye on other OSs, and RaspberryPiOS on various hardware.
> Can you run command:
> netstat -apt
> and report the result?
> 
> If not, my suggestion would be to start clean on a different USB or SDCard (save this setup)
> install RPiOS, update everything, and then just install motionEye ONLY.
> This would eliminate the BLYNK as being interfering, then re-add BLYNK after confirming motionEye is working.
> 
> As to the freaky text formatting, MarkUp (here in Github) thinks '#' as the first character on a line designates
> 
> # BOLD

here you are:
`pi@raspberrypi:~ $ netstat -apt
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:netbios-ssn     0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:tproxy          0.0.0.0:*               LISTEN      -
tcp        0      0 localhost:8082          0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:8083            0.0.0.0:*               LISTEN      -
tcp        0      0 localhost:5939          0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:8085            0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:ssh             0.0.0.0:*               LISTEN      -
tcp        0      0 localhost:postgresql    0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:smtp            0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:8765            0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:microsoft-ds    0.0.0.0:*               LISTEN      -
tcp        0      0 localhost:7999          0.0.0.0:*               LISTEN      -
tcp     3493      0 192.168.0.10:34750      192.168.0.19:1025       ESTABLISHED -
tcp        0      0 192.168.0.10:8765       192.168.10.1:1028       ESTABLISHED -
tcp        0      0 localhost:33186         localhost:8085          ESTABLISHED -
tcp        0      0 192.168.0.10:58972      dr-streamer.united:http ESTABLISHED -
tcp        0      0 localhost:34412         localhost:5939          ESTABLISHED 962/TeamViewer
tcp       37      0 localhost:8085          localhost:33186         ESTABLISHED -
tcp        0      0 192.168.0.10:34128      192.168.0.24:http       ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35594         ESTABLISHED -
tcp        0      0 192.168.0.10:8765       192.168.10.1:1030       ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35608         ESTABLISHED -
tcp        0      0 192.168.0.10:56218      IE-DUB-ANX-R005.te:5938 ESTABLISHED -
tcp        0      0 192.168.0.10:34380      192.168.0.:microsoft-ds ESTABLISHED -
tcp        0      0 192.168.0.10:58420      RS-BEG-ANX-R005.te:5938 TIME_WAIT   -
tcp        0      0 localhost:postgresql    localhost:35602         ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35586         ESTABLISHED -
tcp        0      0 192.168.0.10:51430      192.168.0.19:http       ESTABLISHED -
tcp        0      0 192.168.0.10:47834      192.168.0.18:rtsp       ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35592         ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35542         ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35588         ESTABLISHED -
tcp        0      0 localhost:5939          localhost:36754         TIME_WAIT   -
tcp        0      0 localhost:5939          localhost:34412         ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35600         ESTABLISHED -
tcp        0      0 192.168.0.10:ssh        192.168.0.100:9847      ESTABLISHED -
tcp        0      0 192.168.0.10:58978      dr-streamer.united:http ESTABLISHED -
tcp        0      0 192.168.0.:microsoft-ds 192.168.0.100:1037      ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35598         ESTABLISHED -
tcp        0      0 localhost:postgresql    localhost:35582         ESTABLISHED -
tcp        0      0 192.168.0.10:8765       192.168.10.1:1029       ESTABLISHED -
tcp6       0      0 [::]:9443               [::]:*                  LISTEN      -
tcp6       0      0 [::]:netbios-ssn        [::]:*                  LISTEN      -
tcp6       0      0 [::]:http-alt           [::]:*                  LISTEN      -
tcp6       0      0 [::]:ssh                [::]:*                  LISTEN      -
tcp6       0      0 [::]:8440               [::]:*                  LISTEN      -
tcp6       0      0 localhost:postgresql    [::]:*                  LISTEN      -
tcp6       0      0 [::]:smtp               [::]:*                  LISTEN      -
tcp6       0      0 [::]:microsoft-ds       [::]:*                  LISTEN      -
tcp6       0      0 localhost:35608         localhost:postgresql    ESTABLISHED -
tcp6       0      0 192.168.0.10:http-alt   192.168.10.5:1026       ESTABLISHED -
tcp6       0      0 localhost:35542         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35582         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35598         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35594         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35586         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35588         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35600         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35592         localhost:postgresql    ESTABLISHED -
tcp6       0      0 localhost:35602         localhost:postgresql    ESTABLISHED -
`

**Comment by starbasessd** [comment]
Yours:
tcp 0 0 localhost:7999 0.0.0.0:* LISTEN 
Mine:
tcp        0      0 0.0.0.0:7999            0.0.0.0:*               LISTEN      2242/motion
----
Something is broken. It should be listening on 0.0.0.0 (any IP address, Eth0, Wlan0, Lo) but only seems to be listening on Lo.
At this point I vote for save your current setup, Install RPiOS fresh on a USB Thumb Drive, then motionEye, and see if you can get it working there.

**Comment by enry86cami** [comment]
> ## Yours:
> tcp 0 0 localhost:7999 0.0.0.0:* LISTEN
> Mine:
> tcp 0 0 0.0.0.0:7999 0.0.0.0:* LISTEN 2242/motion
> Something is broken. It should be listening on 0.0.0.0 (any IP address, Eth0, Wlan0, Lo) but only seems to be listening on Lo.
> At this point I vote for save your current setup, Install RPiOS fresh on a USB Thumb Drive, then motionEye, and see if you can get it working there.

I'm scared about this... 
If I try to remove and install again motioneye?

**Comment by starbasessd** [comment]
Don't use your current SDCard or USB stick. Use a new (different) one. Keep the current one to switch back to.
Uninstalling and re-installing on the same SDCard or USB stick has a much worse chance of surviving.

**Comment by enry86cami** [comment]
> Don't use your current SDCard or USB stick. Use a new (different) one. Keep the current one to switch back to.
> Uninstalling and re-installing on the same SDCard or USB stick has a much worse chance of surviving.

many thanks for your support and patience. I'll schedule this new installation soon...

**Comment by enry86cami** [comment]
> Don't use your current SDCard or USB stick. Use a new (different) one. Keep the current one to switch back to.
> Uninstalling and re-installing on the same SDCard or USB stick has a much worse chance of surviving.

I checked the configuration and I missed a part.... (motion.conf). So it was only necessary:

edit /etc/motioneye/motioneye.conf and /etc/motioneye/motion.conf:
In  /etc/motioneye/motion.conf  Change  webcontrol_localhost on  To  **webcontrol_localhost off**
In /etc/motioneye/motioneye.conf Change  motion_control_localhost true  To  **motion_control_localhost false** 

Many thanks again!

**Comment by tonicxx** [comment]
Hi, i try run script about motion control by PIR connected to the Raspberry Pi Zero W (the same effect on two diffrent Pi and SD Card):
https://github.com/avanc/motioneye-pir
and i have problem because my raspberry still rebooting after i add options: motion_control_port 7999 to the motioneye.conf
and i cant check if script working correctly, below is my configuration an logs, maybe someone have some idea where is the problem:

motion.conf:
webcontrol_interface 1
setup_mode off
camera camera-1.conf
webcontrol_parms 2
webcontrol_localhost off
webcontrol_port 7999

motioneye.conf:
conf_path /data/etc
run_path /tmp
log_path /var/log
media_path /data/output
motion_binary /usr/bin/motion
log_level debug
listen 0.0.0.0
port 8765
mount_check_interval 300
motion_check_interval 10
motion_restart_on_errors false
motion_control_localhost true
motion_control_port 7999
cleanup_interval 43200
remote_request_timeout 10
mjpg_client_timeout 10
mjpg_client_idle_timeout 10
smb_shares false
smb_mount_root /data/media
wpa_supplicant_conf /data/etc/wpa_supplicant.conf
local_time_file /data/etc/localtime
enable_reboot false
enable_update true
smtp_timeout 60
list_media_timeout_email 10
zip_timeout 500
timelapse_timeout 500
add_remove_cameras true
http_basic_auth false
password_hook /usr/libexec/meyepasswd

[motion.log](https://github.com/ccrisan/motioneyeos/files/7382757/motion.log)
[motioneye.log](https://github.com/ccrisan/motioneyeos/files/7382816/motioneye.log)
[dmesg.log](https://github.com/ccrisan/motioneyeos/files/7382841/dmesg.log)
[messages.log](https://github.com/ccrisan/motioneyeos/files/7382966/messages.log)
[camera-1.conf.log](https://github.com/ccrisan/motioneyeos/files/7382987/camera-1.conf.log)

many thanks

**Comment by starbasessd** [comment]
In motioneye.conf:
motion_control_localhost true
needs to be set to  false

**Comment by tonicxx** [comment]
the same effect, any other idea?

**Comment by starbasessd** [comment]
@tonicxx what do you get when you run the command:
 netstat -tulpn
I get for TCP:
[root@meye-4cec78b8 ~]# netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      641/python
tcp        0      0 0.0.0.0:8082            0.0.0.0:*               LISTEN      657/motion
tcp        0      0 0.0.0.0:21              0.0.0.0:*               LISTEN      561/proftpd: (accep
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      546/sshd: /usr/sbin
tcp        0      0 0.0.0.0:445             0.0.0.0:*               LISTEN      582/smbd
tcp        0      0 0.0.0.0:7999            0.0.0.0:*               LISTEN      657/motion
tcp        0      0 0.0.0.0:139             0.0.0.0:*               LISTEN      582/smbd

This line:
tcp        0      0 0.0.0.0:7999            0.0.0.0:*               LISTEN      657/motion
indicates that motion is properly listening on port 7999 for commands.
If you use a browser, what do you see? 
http://ip_address:7999
If you don't see something like:
Motion 4.3.1 Running [1] Camera 
1 
If not, something is corrupted, and it might be faster to save your
camera settings, and re-load the SD Card. If you use dev20201026, you can boot from
a USB stick instead of the SD Card on Pi3B+  and newer.

**Comment by tonicxx** [comment]
i get:
tcp        0      0 0.0.0.0:139             0.0.0.0:*               LISTEN      1087/smbd
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      1147/python
tcp        0      0 0.0.0.0:8081            0.0.0.0:*               LISTEN      1168/motion
tcp        0      0 0.0.0.0:21              0.0.0.0:*               LISTEN      1065/proftpd: (acce
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1049/sshd
tcp        0      0 0.0.0.0:445             0.0.0.0:*               LISTEN      1087/smbd
tcp        0      0 0.0.0.0:7999            0.0.0.0:*               LISTEN      1168/motion

Pi Zero W restarting after i add line "motion_control_port 7999" to motioneye.conf, if i remove them Pi Zero W working correctly, i try use 4 diffrent SD card with new installation and the same effect, any idea?

**Comment by starbasessd** [comment]
What do you get if you open a browser and go to 
http://ip_address:7999

**Comment by tonicxx** [comment]
i see:
Motion 4.2.2+gitUNKNOWN Running [1] Camera 1

**Comment by starbasessd** [comment]
Motion is watching the port correctly.
Your issue lies with the python scripting, and should be addressed to the creator of the scripting add-on.
